apps/src/aichat/redux/slice.ts

```diff
@@ -55,6 +60,29 @@ const aichatSlice = createSlice({
     ) => {
       state.studentChatHistory = action.payload;
     },
+    setOwnChatHistory: (state, action: PayloadAction<ServerChatEvent[]>) => {
+      // Find the last index of an event that marks the start a new conversation with the model.
+      const events = action.payload;
+      let lastResetIndex = -1;
+      for (let i = events.length - 1; i >= 0; i--) {
```

Comment:
Wish we could use `findLastIndex` here, but alas, not supported by all browsers we support and we don't have a polyfill. More discussion here: https://codedotorg.slack.com/archives/C046G4TRLEN/p1741809525076269